File: 0000-LeetCode/Cpp0793-preimage-size-of-factorial-zeroes-function/0793-preimage-size-of-factorial-zeroes-function.cpp
class Solution {
public:
    int preimageSizeFZF(int K) {
        int sum[13]={1};
        for (int i=1; i<13; i++) sum[i]=sum[i-1]*5+1;
        for (int i=12; i>=0; i--) {
            if (K/sum[i]==5) return 0;
            K=K%sum[i];
        }
        return 5;
    }
};

File: 0000-LeetCode/Java/0793-preimage-size-of-factorial-zeroes-function/0793-preimage-size-of-factorial-zeroes-function.java
class Solution {
    public int preimageSizeFZF(int k) {
        long low = 0;
        long high = (long)Math.pow(10,10);
        
        while(low<=high){
            long mid  = low + (high-low)/2;
            long val = mid;
            long ans = 0;
            while(val!=0){
                val/=5;
                ans+=val;
            }
            if(ans==k){
                return 5;
            }else if(ans>k){
                high = mid-1;
            }else{
                low = mid+1;
            }
        }
        
        return 0;
    }
}